skbio/io/_base.py

```diff
@@ -185,3 +186,25 @@ def _format_fasta_like_records(generator, id_whitespace_replacement,
                 "scores associated with it." % cardinal_to_ordinal(idx + 1))
 
         yield header, seq.sequence, seq.quality
+
+
+def _line_generator(fh, skip_blanks=False):
+    for line in fh:
+        line = line.strip()
+        if line or not skip_blanks:
+            yield line
+
+
+def _too_many_blanks(fh, max_blanks):
+    count = 0
+    too_many = False
+    for line in _line_generator(fh):
+        if line:
+            break
+        else:
+            count += 1
+            if count > max_blanks:
+                too_many = True
+                break
+    fh.seek(0)
```

Comment:
@ebolyen suggested having this function return the first non-blank line (if there aren't too many blanks), then the sniffers can use this line (and the rest of the contents of the file handle) to continue sniffing. Having a `seek(0)` call is unsafe in sniffers because the user may supply a file handle that has already been `seek`ed to an offset.

Related to this, I noticed that the current FASTA sniffer is already calling `seek(0)` before trying sniff as QUAL. This bug would be great to fix in this PR since we're already changing the logic of the sniffer somewhat. @ebolyen suggested a new sniffing strategy:

1. try to read up to 10 FASTA records
2. for each record, check the contents of the sequence to see if they look like quality scores. if they are, bail since we don't want to sniff a FASTA file as QUAL